### PR TITLE
import into: add analyze table for import into

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref #42930 

Problem Summary:

### What is changed and how it works?
- add analyze table in post progress
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/realtikvtest/loaddatatest/load_data_test.go [hunk 1]
```diff
@@ -1128,3 +1129,33 @@ func (s *mockGCSSuite) TestDiskQuota() {
...
+
+	// without analyze, use idx_ac
+	s.tk.MustExec("SET GLOBAL tidb_enable_auto_analyze=ON;")
+	s.tk.MustQuery("EXPLAIN SELECT * FROM load_data.analyze_table WHERE a=1 and b=1 and c=1;").CheckContain("idx_ac(a, c)")
+	s.tk.MustQuery("SHOW ANALYZE STATUS;").CheckNotContain("analyze_table")
+
+	sql := fmt.Sprintf(`IMPORT INTO load_data.analyze_table FROM 'gs://test-load/analyze-1.tsv?endpoint=%s'`, gcsEndpoint)
+	s.tk.MustExec(sql)
+	require.Eventually(s.T(), func() bool {
+		result := s.tk.MustQuery("EXPLAIN SELECT * FROM load_data.analyze_table WHERE a=1 and b=1 and c=1;")
```
Comment (lance6716) [inline tests/realtikvtest/loaddatatest/load_data_test.go:1157]: please directly check stat is updated

File: executor/asyncloaddata/progress.go [hunk 2]
```diff
@@ -73,6 +79,27 @@ func NewProgress(logicalImport bool) *Progress {
...
+func (p *Progress) AddColSize(colSizeMap map[int64]int64) {
+	p.colSizeMu.Lock()
+	defer p.colSizeMu.Unlock()
+	for key, value := range colSizeMap {
+		p.colSizeMap[key] += value
+	}
+}
+
+// GetColSize returns the size of the column.
+func (p *Progress) GetColSize() map[int64]int64 {
```
Comment (lance6716) [inline executor/asyncloaddata/progress.go:92]: maps.Clone

File: executor/import_into.go [hunk 3]
```diff
@@ -159,5 +168,21 @@ func (e *ImportIntoExec) doImport(distImporter *loaddata.DistImporter, task *pro
...
 		userStmtCtx.SetAffectedRows(importResult.Affected)
 	}
+	if err2 := flushStats(ctx, se, e.importPlan.TableInfo.ID, distImporter.Result(task)); err2 != nil {
+		logutil.BgLogger().Error("flush stats failed", zap.Error(err2))
+	}
 	return err
 }
+
+// flushStats flushes the stats of the table.
+func flushStats(ctx context.Context, se sessionctx.Context, tableID int64, result importer.JobImportResult) error {
```
Comment (lance6716) [inline executor/import_into.go:178]: we can add GetStatsCollector to session and use statsCollector.Update? that's more straightforward

Comment (GMHDBJD) [reply executor/import_into.go:178]: don't want to deal with import circle... https://github.com/pingcap/tidb/blob/f098548642c3f05d6e11c54bd3012b9f58cfb3a5/sessionctx/context.go#L129

File: tests/realtikvtest/loaddatatest/load_data_test.go (hunk 1 above)
Comment (GMHDBJD) [reply tests/realtikvtest/loaddatatest/load_data_test.go:1157]: L1160

**Review by lance6716 — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]
lgtm

**Comment by D3Hunter** [comment]
let's wait https://github.com/pingcap/tidb/pull/44602 merged, and rebase together

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**